src/Counter.elm

```diff
@@ -0,0 +1,97 @@
+module Counter exposing (main)
+
+import Browser
+import Html exposing (button, div, h2, input, strong, text)
+import Html.Attributes as Attributes
+import Html.Events exposing (onClick, onInput)
+
+
+main =
+    Browser.sandbox
+        { init = init
+        , update = update
+        , view = view
+        }
+
+
+type alias Model =
+    { counter : Float
+    , content : String
+    , lastCounter : Float
+    }
+
+
+init : Model
+init =
+    { counter = 0, content = "0", lastCounter = 0 }
+
+
+type Msg
+    = Plus
+    | Minus
+    | Multiply
+    | Div
+    | ChangeNumber String
+    | ResetCounter
+    | Undo
+
+
+update : Msg -> Model -> Model
+update msg model =
+    let
+        userInputNum =
+            Maybe.withDefault 0 (String.toFloat model.content)
+    in
+    case msg of
+        Plus ->
+            { model
+                | counter = model.counter + userInputNum
+                , lastCounter = model.counter
+            }
+
+        Minus ->
+            { model
+                | counter = model.counter - userInputNum
+                , lastCounter = model.counter
+            }
+
+        Multiply ->
+            { model
+                | counter = model.counter * userInputNum
+                , lastCounter = model.counter
+            }
+
+        Div ->
+            { model
+                | counter = model.counter / userInputNum
+                , lastCounter = model.counter
+            }
+
+        ChangeNumber newNumber ->
+            { model | content = newNumber }
+
+        ResetCounter ->
+            { model
+                | counter = 0
+                , lastCounter = model.counter
+            }
+
+        Undo ->
+            { model
+                | counter = model.lastCounter
+                , lastCounter = model.counter
+            }
```

Comment:
This means that multiple consequent `undo`s just toggle the 2 most recent counters.
If you want a challenge, you can implement `undo` throughout the whole history 😃